content/ConnectingToDatabase/ConnectingToDatabase.q.md

```diff
@@ -0,0 +1,837 @@
+
+- Undefine variable: `db` on line 4
+
+- Undefined variable: `db` on line 9
+
+- Unable to establish a connection.
+
+- Undefined variable: `conn` on line 13
+
+- syntax error, unexpected `=`, expecting end of file on line 9
+
+:::
+
+
+:::
+
+/// type=REPL readonly=true
+
+```php
+<?php
+    $host = 'newhost';
+    $db = 'MyDatabase';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+
+    try {
+        $conn = new PDO("pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password");
+        if ($conn) {
+            echo "Successfully connected!"; 
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection.";
+    }
+?>
+```
+/// type=SS, answer=[4]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Successfully connected!`.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `"Unable to establish a connection."`.
+
+
+/// type=SS, answer=[3]
+
+Which statement correctly describes the error?
+
+- There is no DSN prefix `psql:` on line 9.
+
+- On line 9, the DSN parameters are separated by a semicolon `;`.
+
+- The database server's hostname `newhost` on line 1 is incorrect.
+
+- There is no open curly brace `{` after the `if` statement on line 10.
+
+- On line 9, the command `"pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password"` is passed as an argument.
+
+:::
+
+
+/// type=CR, answer=[tests/ConnectingToDatabase/IncorrectHostNameTest.php]
+
+Correct the code so that it outputs the string `Successfully connected!` using its database server’s hostname as `localhost`.
+
+```php
+<?php
+    $host = 'newhost';
+    $db = 'MyDatabase';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+
+    try {
+        $conn = new PDO("pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password");
+        if ($conn) {
+            echo "Successfully connected!"; 
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection.";
+    }
+?>
+```
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+   $host = 'localhost';
+   $db = 'MyDatabase';
+   $port = '5432';
+   $username = 'postgres';
+   $password = 'Admin01';
+   
+   $dsn = "pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password";
+
+   try {
+       $conn = new PDO($dsn);
+       if ($conn) {
+           echo "Successfully connected!";
+       }
+   } catch (Exception $e) {
+       echo "Unable to establish a connection.";
+   }
+?>
+```
+/// type=SS, answer=[3]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Successfully connected!`.
+
+- It prints `"Successfully connected!"`.
+
+- It prints `Unable to establish a connection.`.
+
+
+/// type=SS, answer=[4]
+
+On line 4, replace the string value `5432` with `1234`. Execute the program. What is its output?
```

Comment:
`string value` is confusing here... I think we need to choose whether we use `string` or `value` here... Please check. Thanks.